packages/server/routes/api/guilds.js

```diff
@@ -4,51 +4,166 @@ const router = express.Router();
 const GuildController = require("../../controllers/guilds");
 const AuthController = require("../../controllers/auth");
 
+// Get all guilds from database
 router.get("/", AuthController.authenticate, async (request, response) => {
   try {
-    const guilds = await GuildController.getAllGuilds();
     response.status(200).json({
       status: "success",
       data: {
-        guilds,
-      },
+        guilds: await GuildController.getAllGuilds(),
+      }
     });
   } catch (error) {
-    response.status(500).json({
+    // Should never error.
+    // In the case that there are no guilds, 
+    // the return would be an empty array.
+    // Here for redudancy.
+    response.status(400).json({
       status: "error",
-      message: error.message,
+      errorName: error.name,
+      errorMessage: error.message,
     });
   }
 });
 
+// Get guild by ID
 router.get(
   "/:guildId",
   AuthController.authenticate,
   async (request, response) => {
     try {
-      const { guildId } = request.params;
 
-      if (guildId === undefined) {
-        return response.status(400).json({
-          status: "fail",
-          data: {
-            guildId: "Guild ID not provided",
-          },
-        });
+      return response.status(200).json({
+        status: "success",
+        data: {
+          guild: await GuildController.getGuild(request.params.guildId),
+        }
+      });
+      
+    } catch (error) {
+      switch (error.name) {
+
+        case "PrismaClientKnownRequestError":
+          return response.status(400).json({
+            status: "error",
+            errorName: error.name,
+            errorMessage: "The guildId Provided had invalid format.",
+          });
+          
+        case "NotFoundError":
+          return response.status(500).json({
+            status: "fail",
+            errorName: error.name,
+            errorMessage: error.message,
+          });
+        
+        // There are only two known error cases.
+        // This default acts as a redundant check if an unknown error occurs.
+        default:
+          return response.status(500).json({
+            status: "fail",
+            errorName: error.name,
+            errorMessage: error.message,
+          });
       }
+    }
+  }
+);
 
-      const guild = await GuildController.getGuild(guildId);
-      response.status(200).json({
+// Create Guild
+router.post(
```

Comment:
Right now, you have `PUT` for updating guilds and `POST` for creating guilds. It should be the other way around, `PUT` is associated with creating new resources in REST API convention, and `POST` is associated with updating resources.

Reply:
Please double check this comment.